test/e2e/validation_test.go

```diff
@@ -393,3 +400,29 @@ func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.Certificate
 	}
 	return nodeCSRs, nil
 }
+
+// validateUpgradeableCondition ensures that the operator's Upgradeable condition is correctly communicated to OLM
+func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatus) error {
+	ocName, present := os.LookupEnv(condition.OperatorConditionName)
+	if !present {
+		// Implies operator is not OLM-managed
+		return nil
+	}
+
+	err := wait.Poll(10*time.Second, time.Minute, func() (bool, error) {
```

Comment:
Is it possible to use some of the existing constants?

Reply:
Refactored to reduce code duplication.